cmd/main.go

```diff
@@ -147,11 +149,10 @@ func main() {
 
 	setupLog.Info("enabling features", "FeatureGate", metal3iov1alpha1.CurrentFeatureGate.String())
 
-	if versionInfo.InstalledVersion != "" {
-		if err := metal3iov1alpha1.ValidateVersion(versionInfo.InstalledVersion); err != nil {
-			setupLog.Error(err, "invalid ironic-version")
-			os.Exit(1)
-		}
+	versionInfo, err := ironic.NewVersionInfo(ironicImages, ironicVersion, databaseImage)
+	if err != nil {
+		setupLog.Error(err, "invalid ironic-version")
```

Comment:
```suggestion
		setupLog.Error(err, "invalid ironic-version: %q", ironicVersion)
```
Personally I prefer invalid values shown in error messages.

Reply:
You'll get it through `err`, the initial error message already includes the invalid value